pulse2percept/retina.py

```diff
@@ -901,20 +997,337 @@ def dva2ret(r_deg):
+    """Determines the contribution of a single axon to the current map
+
+    This function determines the contribution of a single axon to the current
+    map based on a sensitivity rule (i.e., how the activation threshold of the
+    axon differs as a function of distance from the soma), and an contribution
+    rule (i.e., how the different activation thresholds along the axon are
+    combined to determine the axon contribution).
+
+    Parameters
+    ----------
+    axon_dist : tuple (indices, distances)
+        A tuple containing a list of coordinates (indices into the retinal
+        coordinates mesh grid) and distances for each axon segment.
+    current_spread : 2D array
+        A 2D current spread map that must have the same dimensions as the
+        `xg`, `yg` meshgrid.
+    sensitivity_rule : {'decay', 'Jeng2011'}, optional, default: 'decay'
+        This rule specifies how the activation of the axon differs as a
+        function of distance from the soma. The following options are
+        available:
+        - 'decay':
+            Axon sensitivity decays exponentially with distance. Specify
+            `decay_const` to change the steepness of the fall-off with
+            distance.
+        - 'Jeng2011':
+            Axon sensitivity peaks near the sodium band (50um from the soma),
+            then plateaus on the distal axon at roughly half of the peak
+            sensitivity. See Figure 2 in Jeng, Tang, Molnar, Desai, and Fried
+            (2011). The sodium channel band shapes the response to electric
+            stimulation in retinal ganglion cells. J Neural Eng 8 (036022).
+    contribution_rule : {'max', 'sum', 'power-mean'}, optional, default: 'max'
+        This rule specifies how the activation thresholds across all axon
+        segments are combined to determine the contribution of the axon to the
+        current spread. The following options are available:
+        - 'max':
+            The axon's contribution to the current spread is equal to the max.
+            sensitivity across all axon segments.
+        - 'sum':
+            The axon's contribution to the current spread is equal to the sum
+            sensitivity across all axon segments.
+        - 'power-mean':
+            The axon's contribution to the current spread is equal to the mean
+            sensitivity across all axon segments. Specify `powermean_exp` to
+            change the exponent of the generalized (power) mean, calculated as
+            np.mean(x ** powermean_exp) ** (1.0 / powermean_exp). Default is 1,
+            which is equal to the arithmetic mean.
+    min_contribution : float, optional, default: 0.01
+        Current contributions below this value will not be counted.
+    decay_const : float, optional, default: 2.0
+        When `sensitivity_rule` is set to 'decay', specifies the decay constant
+        of the exponential fall-off.
+    powermean_exp : float, optional, default: 1.0
```

Comment:
You might want to add error handling to catch the case where the user is trying to change `powermean_exp`, but the `contribution_rule` is not set to `mean`. I would do this by setting the default value of the argument to `None` and then checking if it is set to anything else on input, relative to `contribution_rule`